interpreter/cling/lib/Interpreter/DynamicLibraryManagerSymbol.cpp

```diff
@@ -345,15 +345,13 @@ std::string cached_realpath(llvm::StringRef path, llvm::StringRef base_path = ""
 
   // Handle path list items
   for (auto item : p) {
-    if (item.startswith(".")) {
-      if (item == "..") {
-        size_t s = result.rfind(sep);
-        if (s != llvm::StringRef::npos) result.resize(s);
-        if (result.empty()) result = sep;
-      }
-      continue;
-    } else if (item == "~") {
-      continue;
+    if (item == ".")
+      continue; // skip "." element in "abc/./def"
+    if (item == "..") {
+      // collapse "a/b/../c" to "a/c"
+      size_t s = result.rfind(sep);
+      if (s != llvm::StringRef::npos) result.resize(s);
+      if (result.empty()) result = sep;
```

Comment:
@Axel-Naumann , I think you have missed `continue;` here . I am testing this in cmssw now and will let you know how it goes

Reply:
YES! :1st_place_medal: Thank you so much, that's totally it! I'll add a test for this.

Reply:
I confirm that after adding `continue;` cmssw tests passed

Reply:
Merged as https://github.com/root-project/root/pull/9951 - thanks, @smuzaffar !